.github/workflows/publish.yml

```diff
@@ -0,0 +1,29 @@
+name: Publish to PyPi
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  build-and-publish:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout source code
+      uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.x"
+    - name: Install dependencies
+      run: python -m pip install --upgrade pip setuptools wheel
+
+    - name: Build package
+      run: python setup.py sdist bdist_wheel
+
+
+    - name: Publish package to PyPI
+      env:
+          TWINE_USERNAME: "__token__"
+          TWINE_PASSWORD: ${{ secrets.PYPI_API_TOKEN }}
+      run: python -m pip install --upgrade twine && twine upload dist/*
```

Comment:
It would be nice to test the distribution before publishing it. I am not sure if `twine` can do it automatically. Even if it is possible it probably doesn't happen because tests are not part of the package directory tree. In this case I would suggest running the tests manually before publishing.